internal/pubsub/pubsub.go

```diff
@@ -136,7 +142,12 @@ func (ps *PubSub[T, M]) SubscribeJSON(mask M, subCh chan<- []byte, doneCh <-chan
 				if err != nil {
 					break
 				}
-				subCh <- append(GetByteBuffer()[:0], buf.Bytes()...)
+
+				select {
+				case subCh <- append(GetByteBuffer()[:0], buf.Bytes()...):
+				case <-doneCh:
+					break loop
```

Comment:
```suggestion
return
```

We can just return here right?

Reply:
There is some logic after the loop... I was annoyed as well.... Maybe we should defer that logic.

Reply:
Rewrote it. PTAL